runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/BlockManagerWorker.java

```diff
@@ -367,27 +396,47 @@ public void removeBlock(final String blockId,
    * @throws InvalidProtocolBufferException from errors during parsing context descriptor
    */
   public void onOutputContext(final ByteOutputContext outputContext) throws InvalidProtocolBufferException {
+    LOG.info("dongjoo: BMW onOutputContext, outputcontext {}", outputContext);
     final ControlMessage.BlockTransferContextDescriptor descriptor =
       ControlMessage.BlockTransferContextDescriptor.PARSER.parseFrom(outputContext.getContextDescriptor());
     final DataStoreProperty.Value blockStore = convertBlockStore(descriptor.getBlockStore());
     final String blockId = descriptor.getBlockId();
     final KeyRange keyRange = SerializationUtils.deserialize(descriptor.getKeyRange().toByteArray());
-
+    LOG.info("Dongjoo, descriptor, blockstore, blockid keyrange finalized, before run");
     backgroundExecutorService.submit(new Runnable() {
       @Override
       public void run() {
         try {
           final Optional<Block> optionalBlock = getBlockStore(blockStore).readBlock(blockId);
+          final Optional<Block> optionalSpilledBlock = getSpilledBlock(blockId);
```

Comment:
Here and all below, I couldn't really understood the need for `optionalSpilledBlock`. Is it disjoint from `optionalBlock`? 
Is a Block moves from `optionalBlock` to `optionalSpilledBlock` when it is spilled?
Is there any consistency issues?(looks like there are no locks) e.g. a Block was in `optionalBlock` of Executor 1 -> Executor 2 read it -> Executor 1 moved the block to `optionalSpilledBlock`?